primitives/runtime/src/transaction_validity.rs

```diff
@@ -161,6 +161,35 @@ impl Into<TransactionValidity> for UnknownTransaction {
 	}
 }
 
+/// The origin of the transaction.
+///
+/// Depending on the source we might apply different validation schemes.
+/// For instance we can disallow specific kinds of transactions if they were not produced
+/// by our local node (for instance off-chain workers).
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, parity_util_mem::MallocSizeOf)]
+pub enum TransactionSource {
+	/// Transaction is already included in block.
+	///
+	/// This means that we can't really tell where the transaction is coming from,
+	/// since it's already in the received block. Note that the custom validator
+	/// using either `Local` or `External` should most likely just allow `InBlock`
+	/// transactions as well.
+	InBlock,
+
+	/// Transaction is coming from a local source.
+	///
+	/// This means that the transaction was produced either internally by the node
+	/// (for instance an Off-Chain Worker, or an Off-Chain Call) opposed
+	/// to being received over the network.
+	Local,
+
+	/// Transaction has been received externally.
+	///
+	/// This means the transaction has been received from (usually) "untrusted" source,
+	/// for instance received over the network or RPC.
+	External,
```

Comment:
Maybe also differentiate Network/RPC ?

Reply:
I was thinking about it, but I don't feel like `RPC` is strong enough use case to separate it. The way I see it the transactions received over RPC are either `External` if you are running a public node or `Local` if you are the only user of the RPC. What would be the third use case?

As mentioned in the description I'd rather see a CLI option that allows you to mark RPC-received transactions as `Local` (if you are not exposing your node externally).

Reply:
Yeah, I had this case in mind, where you just have your own node and want to prioritise your own transactions you submit via RPC.

CLI option will work great for that.

Reply:
Is there already an issue for this?